src/cargo/ops/lockfile.rs

```diff
@@ -19,20 +20,25 @@ pub fn load_pkg_lockfile(pkg: &Package) -> CargoResult<Option<Resolve>> {
 
 pub fn load_lockfile(path: &Path, sid: &SourceId) -> CargoResult<Option<Resolve>> {
     // If there is no lockfile, return none.
-    let mut f = match File::open(path) {
-        Ok(f) => f,
+    let s = match load_string(path) {
```

Comment:
Yeah that's plausible, although it may return a `CargoResult` which could make it difficult to get at the underlying `io::Error`, but so long as it preserves the original semantics I'm fine with whatever!